src/main/java/io/lettuce/core/cluster/models/tracking/TrackingInfoParser.java

```diff
@@ -0,0 +1,75 @@
+package io.lettuce.core.cluster.models.tracking;
+
+import io.lettuce.core.protocol.CommandKeyword;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Parser for Redis <a href="https://redis.io/docs/latest/commands/client-trackinginfo/">CLIENT TRACKINGINFO</a> command output.
+ *
+ * @author Tihomir Mateev
+ * @since 7.0
+ */
+public class TrackingInfoParser {
+
+    /**
+     * Utility constructor.
+     */
+    private TrackingInfoParser() {
+    }
+
+    /**
+     * Parse the output of the Redis CLIENT TRACKINGINFO command and convert it to a {@link TrackingInfo}
+     *
+     * @param trackinginfoOutput output of CLIENT TRACKINGINFO command
+     * @return an {@link TrackingInfo} instance
+     */
+    public static TrackingInfo parse(List<?> trackinginfoOutput) {
+
+        verifyStructure(trackinginfoOutput);
+
+        List<?> flags = (List<?>) trackinginfoOutput.get(1);
+        Long clientId = (Long) trackinginfoOutput.get(3);
+        List<?> prefixes = (List<?>) trackinginfoOutput.get(5);
+
+        Set<TrackingInfo.TrackingFlag> parsedFlags = new HashSet<>();
+        List<String> parsedPrefixes = new ArrayList<>();
+
+        for (Object flag : flags) {
+            String toParse = (String) flag;
+            parsedFlags.add(TrackingInfo.TrackingFlag.from(toParse));
+        }
+
+        for (Object prefix : prefixes) {
+            parsedPrefixes.add((String) prefix);
+        }
+
+        return new TrackingInfo(parsedFlags, clientId, parsedPrefixes);
+    }
+
+    private static void verifyStructure(List<?> trackinginfoOutput) {
```

Comment:
not sure with the choice of verification and parsing. 
Is there any approach which embraces possible future changes on response while still remains safe ?

Reply:
I always try to consume at most elements that I know should exist. That would manifest in `trackinginfoOutput.size() < 6` then throw exception.

Does that help?

Reply:
that,, or checking if all the keys we are looking for exist in response ...
would do, i believe.

Reply:
Yeah, I was not very happy with this part either, to tell you the truth.

I lack experience in how stable this contract is, but it felt to me that future changes **should** break the code.
My reasoning:
- the parser is a convenience and the consumers of the driver could easily parse whatever new response is coming themselves, so we are not blocking them until we provide a new release
- parsing heuristically could end up providing the wrong data to the user, e.g. if the order is changed and some other piece of information is given in the place of what we expect - the app would not fail with an error, but there would be wrong data processed. For example if the server starts returning

```
1) "flags"
2) 1) "off"
3) "some-new-property"
4) "-1"
5) "prefixes"
6) (empty list or set)
7) "redirect"
8) "-1"
``` 
instead of
```
1) "flags"
2) 1) "off"
3) "redirect"
4) "-1"
5) "prefixes"
6) (empty list or set)
7) "some-new-property"
8) "-1"
``` 

I know that this strictness is inconsistent with the other parsers though (`ClusterSlotParser`, `ClusterPartitionParser`), but I thought it is more of an internal implementation thing.

What do you think, folks?

Reply:
The validation can be improved once we change the return type to Map. Once it's done, you can check for the "known" fields in the Map as a validation step without verifying the order, which Redis does not guarantee.

Reply:
FWIW, for some responses, we convert the associative array into a map for easier access to known keys. Nested arrays would go into lists (such as `flags`).